File: TypeScript/Linked-Lists/SingleListNode.ts
/**
 * Representing a Node class in a Singly Linked List
 * @template T the type of data stored in the node.
 * 
 * @property data: the data stored in the node
 * @property next: A reference(pointer) to the next element, by default it is null.
 */
export class ListNode<T> {

    constructor(
        public data: T,
        public next?: ListNode<T>
    ) {}
}

File: Java/LinkedLists/SingleListNode.java
package LinkedLists;

public  class SingleListNode<T> implements IListNode<T> {
    public T data;
    public SingleListNode<T> next;

    public SingleListNode(T data) {
        this.data = data;
    }
    public SingleListNode(T data, SingleListNode<T> next) {
        this.data = data;
        this.next = next;
    }
}
